ddl/ddl_api.go

```diff
@@ -6400,6 +6455,129 @@ func (d *ddl) AlterTableAttributes(ctx sessionctx.Context, ident ast.Ident, spec
 	return errors.Trace(err)
 }
 
+func (d *ddl) AlterTableStatsOptions(ctx sessionctx.Context, ident ast.Ident, spec *ast.StatsOptionsSpec) error {
+	schema, tb, err := d.getSchemaAndTableByIdent(ctx, ident)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	meta := tb.Meta()
+	statsOptions := model.NewStatsOptions()
+	if !spec.Default {
+		if spec.StatsOptions == "" && meta.StatsOptions != nil {
+			statsOptions = meta.StatsOptions
+		} else if spec.StatsOptions != "" {
+			attrBytes := []byte(spec.StatsOptions)
+			var attrs map[string]string
+			err = json.Unmarshal(attrBytes, &attrs)
+			if err != nil {
+				return errors.Wrapf(err, "Stats options should be in json format")
+			}
+			hasBuckets, hasTopn, hasSampleRate, hasColChoice, hasColList := false, false, false, false, false
```

Comment:
So we don't check the legality of the option here? Left the check to the place when we call the analyze?